### PR TITLE
feat: create pagination controller

### PR DESCRIPTION
**Pull request by abdulrahman-2020**
- create pagination controller

Relates #41

### CODE REVIEW

File: server/utils/validation/workIdValidation.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// this file to validate params.id attribute
+const Joi = require('joi');
+
+const workIdValidation = Joi.object({
+  id: Joi.number().integer().positive().required(),
```
Comment (LinaYahya) [inline server/utils/validation/workIdValidation.js:5]:
```suggestion
  id: Joi.number().integer().positive().required(),
  page: // create a page validation though 
```

File: server/controllers/pagenation/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const { Work } = require('../../database/models');
+const { customError } = require('../errors');
+const { workIdValidation } = require('../../utils/validation');
+
+const getPageData = async (req, res, next) => {
+  try {
+    const { providerId } = await workIdValidation.validateAsync(req.params);
+    const { page } = req.query;
+    const pageData = await Work.findAndCountAll({
```
Comment (LinaYahya) [inline server/controllers/pagenation/index.js:9]: Why you're using count here!

**Review by LinaYahya — CHANGES_REQUESTED** [review]
Good Job AbedRahman,
Small changes required and add tests also

File: server/controllers/pagenation/index.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    const user = await User.findByPk(userId);
+    if (!user) {
+      throw customError('User not found', 404);
+    }
+    const pageData = await Work.findAndCountAll({
+      limit: 4,
+      offset: (pageNum - 1) * 4,
+      where: { userId }, // conditions
+    });
+    console.log(pageData);
```
Comment (devstm) [inline server/controllers/pagenation/index.js:20]: remove console.log

File: server/controllers/pagenation/index.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    if (!pageData.count) {
```
Comment (devstm) [inline server/controllers/pagenation/index.js:21]: If there is no works; response status code should be 400 and message  "There Is No Works".
I think 'Page Not Found' not descriptive message

**Review by devstm — CHANGES_REQUESTED** [review]

**Review by RandSohail — APPROVED** [review]

**Review by devstm — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Pull request merged**